modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLECPSSOTestCase.java

```diff
@@ -0,0 +1,223 @@
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.identity.sso.saml.stub.types.SAMLSSOServiceProviderDTO;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.identity.integration.test.util.Utils;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+
+public class SAMLECPSSOTestCase extends AbstractSAMLSSOTestCase {
+
+    private SAMLConfig config;
+    private static final Log log = LogFactory.getLog(AbstractSAMLSSOTestCase.class);
+    private static final String APPLICATION_NAME = "SAML-ECP-TestApplication";
+    private File identityXML;
+    private ServerConfigurationManager serverConfigurationManager;
+    private static final String SAML_ECP_ISSUER = "https://localhost/shibboleth";
+    @Factory(dataProvider = "samlConfigProvider")
+    public SAMLECPSSOTestCase(SAMLConfig config){
+        if (log.isDebugEnabled()){
+            log.info("SAML SSO Test initialized for " + config);
+        }
+        this.config = config;
+    }
+
+    @DataProvider(name = "samlConfigProvider")
+    public static SAMLConfig[][] samlConfigProvider(){
+        return  new SAMLConfig[][]{
+                {new SAMLConfig(TestUserMode.SUPER_TENANT_ADMIN, User.SUPER_TENANT_USER, HttpBinding.HTTP_SOAP,
+                        ClaimType.LOCAL, App.ECP_APP)},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+        super.init(config.getUserMode());
+        changeISConfiguration();
+        super.init(config.getUserMode());
+        super.testInit();
+        super.createUser(config);
+        super.createApplication(config, APPLICATION_NAME);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception{
+
+        super.deleteUser(config);
+        super.deleteApplication(APPLICATION_NAME);
+        super.testClear();
+        resetISConfiguration();
+    }
+
+    @Test(description = "Add service provider", groups = "wso2.is", priority = 1)
+    public void testAddSP() throws Exception {
+        Boolean isAddSuccess = ssoConfigServiceClient.addServiceProvider(super.createECPServiceProviderDTO(config));
+        Assert.assertTrue(isAddSuccess, "Adding a service provider has failed for " + config);
+        SAMLSSOServiceProviderDTO[] samlssoServiceProviderDTOs = ssoConfigServiceClient
+                .getServiceProviders().getServiceProviders();
+        log.debug(samlssoServiceProviderDTOs[0].getIssuer());
```

Comment:
Please add a if clause to check if debug is enabled.

Reply:
> Please add a if clause to check if debug is enabled.

Removed unnecessary debug logs